File: back-end/src/main/java/org/scutsu/materialmanagement/backend/repositories/PersonRepository.java
package org.scutsu.materialmanagement.backend.repositories;

import org.scutsu.materialmanagement.backend.models.Person;
import org.scutsu.materialmanagement.backend.models.PersonProjection;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

@RepositoryRestResource(excerptProjection = PersonProjection.class, path = "people", collectionResourceRel = "people")
public interface PersonRepository extends PagingAndSortingRepository<Person, Long> {
}

File: back-end/src/main/kotlin/org/scutsu/materialmanagement/backend/repositories/PersonRepository.kt
package org.scutsu.materialmanagement.backend.repositories

import org.scutsu.materialmanagement.backend.models.Person
import org.scutsu.materialmanagement.backend.models.PersonMetadata
import org.scutsu.materialmanagement.backend.models.PersonMetadataType
import org.springframework.data.repository.CrudRepository

interface PersonRepository: CrudRepository<Person, Long> {
}

interface PersonMetadataTypeRepository: CrudRepository<PersonMetadataType, Long> {
}
